lib/NormalModuleFactory.js

```diff
@@ -524,11 +527,15 @@ class NormalModuleFactory extends ModuleFactory {
 					}
 				);
 
+				const dependencyType =
+					dependencies.length > 0 ? dependencies[0].category : undefined;
+
 				// resource with scheme
 				if (scheme) {
 					resourceData = {
 						resource: unresolvedResource,
 						data: {},
+						dependencyType,
```

Comment:
This should not be put into the resource data. It's not info about the resource. Just pass `dependencyType` directly to the ruleset